### PR TITLE
Task/152 improvement on string matching

### PR DESCRIPTION
**Pull request by tora-o**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Improvements

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Jedddy — APPROVED** [review]

File: poetry.lock
Comment (CunningBard) [inline poetry.lock:1]: poetry lock?

Comment (CunningBard) [reply poetry.lock:1]: this shouldn't be committed right?

**Review by CunningBard — APPROVED** [review]
LGTM

File: poetry.lock
Comment (CunningBard) [reply poetry.lock:1]: nvm, i forgot you have to

**Review by kylecdrck — APPROVED** [review]
LGTM

**Pull request merged**